pkg/controller/volume/pvcprotection/pvc_protection_controller.go

```diff
@@ -256,6 +259,29 @@ func (c *Controller) pvcAddedUpdated(obj interface{}) {
 	}
 }
 
+func (c *Controller) pvcDeleted(obj interface{}) {
+	tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+		return
+	}
+	pvc, ok := tombstone.Obj.(*v1.PersistentVolumeClaim)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a PVC %#v", obj))
+		return
+	}
+
+	if protectionutil.IsDeletionCandidate(pvc, volumeutil.PVCProtectionFinalizer) {
+		isUsed, err := c.isBeingUsed(pvc)
+		if err != nil {
+			return
+		}
+		if !isUsed {
+			c.removeFinalizer(pvc)
```

Comment:
actually, this doesn't make sense. if the PVC was deleted, then there's no object to remove the finalizer from